types.c

```diff
@@ -48,7 +48,7 @@ fltype_t *get_type(value_t t)
             ft->elsz = eltype->size;
             ft->eltype = eltype;
             ft->init = &cvalue_array_init;
-            eltype->artype = ft;
+            //eltype->artype = ft; -- this is a bad idea since some types carry array sizes
```

Comment:
Thanks for catching this. Maybe this line should be included with a check that the length of `t` is 2?